server/src/main/java/org/opensearch/cluster/SnapshotsInProgress.java

```diff
@@ -747,7 +747,12 @@ public void writeTo(StreamOutput out) throws IOException {
             snapshot.writeTo(out);
             out.writeBoolean(includeGlobalState);
             out.writeBoolean(partial);
-            out.writeByte(state.value());
+            if ((out.getVersion().before(Version.V_3_0_0)) && state == State.PARTIAL) {
```

Comment:
Shouldn't this be `out.getVersion().before(Version.V_2_14_0))`?

Reply:
> Shouldn't this be `out.getVersion().before(Version.V_2_14_0))`?

After backport - yes

Reply:
@owaiskazi19  @reta  There was a comment earlier in TransportSnapshotsStatusAction.java
```
// Translating both PARTIAL and SUCCESS to SUCCESS for now
// TODO: add the differentiation on the metadata level in the next major release
```
So I assumed this change would be only introduced from next major i.e. 3.0.0 release onwards and would not be backported in 2.x branch. Is that view changed now?

Reply:
I don't see any reasons not backporting to 2.x (since we properly handle state for different versions), @aggarwalShivani do you?

Reply:
I think backport to 2.x is a good idea for consistency. @aggarwalShivani  Do we need to update any public documents for this PR?
https://github.com/opensearch-project/documentation-website

Reply:
@ticheng-aws  No, I don't think we need to make any changes in the documentation. The states are already explained in https://github.com/opensearch-project/documentation-website/blob/main/_api-reference/snapshots/get-snapshot-status.md?plain=1#L391